cmd/tg_bot/internal/environment.go

```diff
@@ -23,13 +25,15 @@ var (
 var errUnknownAlertType = errors.New("received unknown alert type")
 
 type TelegramBotEnvironment struct {
-	ChatID int64
-	Bot    *telebot.Bot
-	Mute   bool
+	ChatID        int64
+	Bot           *telebot.Bot
+	Mute          bool
+	pubSubSocket  protocol.Socket
+	subscriptions map[entities.AlertType]string
```

Comment:
I guess these fields are can be shared between multiple goroutines. But I can't find any synchronization here. I think you can easily catch a data race state.

Reply:
I fixed subscriptions' usage, but decided to not touch `mute` as this is not really sensitive data and the case when data race could happen with `mute` is almost 0%

Reply:
But almost is not 0% 🙂. This can lead to unexpected and strange results in some cases. But anyway, if you're ABSOLUTELY SURE that this is ok, then, please, add ATTENTION NOTE about it.